python_modules/dagster/dagster/core/definitions/output.py

```diff
@@ -74,8 +75,10 @@ def __init__(
         metadata=None,
         asset_key=None,
         asset_partitions=None,
-        asset_partitions_def=None
+        asset_partitions_def=None,
```

Comment:
these two args feel odd next to each other, this def arg is undocumented so unclear why its needed